cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -274,11 +385,104 @@ otherwise, the other path will be taken. The diagram for this code would be:
+For workflows that involve forks and joins, a single instance of the workflow token is
+shared by all branches of the fork. Updates to the singleton are made thread-safe through
+synchronized updates, guaranteeing that value you obtain from reading the token is the
+last value written.
+
+Example
+-------
+
+This code sample shows how to obtain values from the token from within a custom action,
+and from within a workflow with a predicate, fork and joins::
+
+  @Override
+  public void run() {
+    ...
+    WorkflowToken token = getContext().getToken();
+    
+    // Set the type of action of the current node:
+    token.put("action_type", "CUSTOM_ACTION");
+ 
+    // Assume that we have the following Workflow: 
+    //                                              |--> PurchaseByCustomer -->|
+    //                                        True  |                          |   
+    // Start --> RecordVerifier --> Predicate ----->|                          |----> StatusReporter --> End    
+    //                                  |           |                          |  |
+    //                                  | False     |--> PurchaseByProduct --->|  |
+    //                                  |                                         |
+    //                                  |------------> ProblemLogger ------------>|
+ 
+    // Use case 1: Predicate can add the key "branch" in the WorkflowToken with value as
+    // "true" if true branch will be executed or "false" otherwise. In "StatusReporter" in
+    // order to get which branch in the Workflow was executed, use:
+    
+    boolean bTrueBranch = token.getAsBoolean("branch");
+ 
+    // Use case 2: You may want to compare the records emitted by "PurchaseByCustomer"
+    // and "PurchaseByProduct", in order to find which job is generating more records:
+    
+    String flattenReduceOutputRecordsCounterName = "org.apache.hadoop.mapreduce.TaskCounter.REDUCE_OUTPUT_RECORDS";
+    String purchaseByCustomerCounterValue = token.get(flattenReduceOutputRecordsCounterName, "PurchaseByCustomer", 
+                                                      WorkflowToken.Scope.SYSTEM);
+    String purchaseByProductCounterValue = token.get(flattenReduceOutputRecordsCounterName, "PurchaseByProduct", 
+                                                     WorkflowToken.Scope.SYSTEM);
+  
+    // Use case 3: Since Workflow can have multiple complex conditions and forks in its
+    // structure, in the "StatusReporter", you may want to know how many actions were
+    // executed as a part of a run. If the number of nodes executed were below a certain
+    // threshold, send an alert. Assuming that every node in the Workflow adds the key
+    // "action_type" with the value as action type for that node in the WorkflowToken,
+    // you can determine the breakdown by action type in a particular Workflow run:
+    
+    List<NodeValueEntry> nodeValues = token.getAll("action_type");
```

Comment:
`action.type`. Basically all occurrences of "action_type" -> "action.type", "startTime" to "start.time",  and "endTime" to "end.time".